src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -359,6 +359,34 @@ public string DefaultPythonEngine
         /// </summary>
         private static string defaultPythonEngine;
 
+        internal event Func<string> RequestUserDataFolder;
```

Comment:
What would happen if there was more than one subscriber to this event?

Reply:
The reason I added an event because this class doesn't have access to the PathManager and I couldn't think of another way to do this. Any ideas are welcome.

Reply:
If the event approach is not ideal, then I can set the path in preference settings by calling a method on it from PreferencesViewModel and passing it the path manager - something like:
```
preferenceSettings.SetSelectedPackagePathForInstall(pathManager);
```

Reply:
I'm okay with either approach, since handler doesn't modify any data and just returns the path I think it's safe even if someone added more subscribers.